### PR TITLE
Fix `MeanSegmentEncoderTransform` to pass inference tests

### LINKED ISSUE #1101: [BUG] Fix `MeanSegmentEncoderTransform` to pass inference tests
### 🐛 Bug Report

According to #1094, `MeanSegmentEncoderTransform` has problems with working on subset of segments and given uncomprehensive error if new segments are met.

### Expected behavior

* `MeanSegmentEncoderTransform` works fine on subset of segments.
* `MeanSegmentEncoderTransform` gives understandable error message if new segments are met.

To make this we should fix the line `df.loc[nan_timestamps, self.idx[:, "segment_mean"]] = self.global_means`. The expression on the right doesn't check segments that are present on the left.

### How To Reproduce

Run #1094.

### Environment

_No response_

### Additional context

_No response_

### Checklist

- [X] Bug appears at the latest library version

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look at #1101.

## Closing issues
Closes #1101.

### CODE REVIEW

File: etna/transforms/encoders/mean_segment_encoder.py [hunk 1]
```diff
@@ -46,9 +50,27 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
...
+            If there are segments that weren't present during training.
         """
+        if self.global_means is None:
+            raise ValueError("The transform isn't fitted!")
+
+        segments = df.columns.get_level_values("segment").unique().tolist()
+        new_segments = set(segments) - self.global_means.keys()
+        if len(new_segments) > 0:
+            raise ValueError(
+                f"This transform can't process segments that weren't present on train data: {new_segments}"
```
Comment (brsnw250) [inline etna/transforms/encoders/mean_segment_encoder.py:68]: Maybe we shouldn't include `new_segments` in error message. If one passes a completely different dataset with a large number of segments, the message will be cluttered.

File: etna/transforms/encoders/mean_segment_encoder.py [hunk 2]
```diff
@@ -46,9 +50,27 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
...
+            )
+
         df = self.mean_encoder.transform(df)
-        segment = df.columns.get_level_values("segment").unique()[0]
+        segments = df.columns.get_level_values("segment").unique()
```
Comment (brsnw250) [inline etna/transforms/encoders/mean_segment_encoder.py:72]: Can we reuse here the `segments` variable from line 64?

File: tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py [hunk 3]
```diff
@@ -22,6 +22,36 @@ def test_mean_segment_encoder_transform(simple_df, transformed_simple_df):
...
+
+
+def test_new_segments_error(simple_df):
+    train_df = simple_df.loc[:, pd.IndexSlice["Moscow", :]]
+    test_df = simple_df.loc[:, pd.IndexSlice["Omsk", :]]
+    transform = MeanSegmentEncoderTransform()
+
+    transform.fit(train_df)
+    with pytest.raises(ValueError, match="This transform can't process segments that weren't present on train data"):
+        _ = transform.transform(test_df)
```
Comment (brsnw250) [inline tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py:52]: Isn't it enough to just call the method without assigning to the variable?

Comment (Mr-Geekman) [reply tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py:52]: May be we should try: https://docs.python.org/3/library/reprlib.html

File: etna/transforms/encoders/mean_segment_encoder.py (hunk 2 above)
Comment (Mr-Geekman) [reply etna/transforms/encoders/mean_segment_encoder.py:72]: Yes, will be fixed.

File: etna/transforms/encoders/mean_segment_encoder.py (hunk 1 above)
Comment (Mr-Geekman) [reply etna/transforms/encoders/mean_segment_encoder.py:68]: May be we should try: https://docs.python.org/3/library/reprlib.html
But I think that in most places we are already printing all broken segments.

File: tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py (hunk 3 above)
Comment (brsnw250) [reply tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py:52]: I've ment assigning to underscore at line 52.

Comment (Mr-Geekman) [reply tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py:52]: I think that in most tests we make assignment to underscore.

File: etna/transforms/encoders/mean_segment_encoder.py (hunk 1 above)
Comment (brsnw250) [reply etna/transforms/encoders/mean_segment_encoder.py]: Ok, let's leave as is right now and consider this case in future.

Comment (Mr-Geekman) [reply etna/transforms/encoders/mean_segment_encoder.py]: I fixed this already.

**Review by brsnw250 — APPROVED** [review]
👍

**Pull request merged**